crates/tako/src/scheduler/state.rs

```diff
@@ -152,55 +151,59 @@ impl SchedulerState {
             );
         }
 
-        for (worker_id, mut task_refs) in task_computes {
-            task_refs.sort_unstable_by_key(|tr| {
-                let t = tr.get();
-                Reverse((t.configuration.user_priority, t.scheduler_priority))
+        for (worker_id, mut task_ids) in task_computes {
+            task_ids.sort_unstable_by_key(|&task_id| {
+                let task = core.get_task(task_id);
+                Reverse((task.configuration.user_priority, task.scheduler_priority))
```

Comment:
Maybe sort_by_cached_key to avoid repeated lookup?

Reply:
Good idea. In the `"scheduler/schedule/tasks=100000, workers=32"` micro-benchmark it seems to be ~5% faster.